### ISSUE TITLE
A BUNCH of Cardcaptor Sakura Clear Card issues

### ISSUE THREAD

**Issue by AshleyAshes** [pymedusa/Medusa#3582]
Running Master on Windows 10.

The file I'm TRYING to get is "[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p]"

First, the obvious issue, using TheTVDB for Cardcaptor Sakura they've decided that Clear Card is the 4th season of Cardcaptor Sakura, rather than it's own series, despite being produced 16-17 years later.  I'm not sure if there's a way around this issue, but using the TVDB as hte indexer with it set to anime results in that episode being understood as ep 1x01 from way back in the late 90s.  Is there even a way to 'fix' this or is TheTVDB just out the window?  (Unless I want to manually rename everything)

Moving on to TVMaze and TMDB.  Both of these indexers DO offer 'Clear Card' as it's own seperate series, super, so 01 is 1x01, that should work.  Now this is where it can't determine episode number.  I -think- this is because the Absolute Number for these series don't exist on this service?  Either way, Medusa is convinced that, with either indexer, the first ep is '0' in terms of absolute numbering.
```
2018-01-07 11:26:19 WARNING  Thread-32 :: [7b9111e] A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p]\[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv: Processing failed: Not enough information to determine what episode this is
2018-01-07 11:26:19 WARNING  Thread-32 :: [7b9111e] Problem(s) during processing, failed for the following files/folders: 
2018-01-07 11:26:19 INFO     Thread-32 :: [7b9111e] Not deleting folder A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p] found the following files: [u'[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv']
2018-01-07 11:26:19 WARNING  Thread-32 :: [7b9111e] Processing failed for A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p]\[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv: Not enough information to determine what episode this is
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Episode doesn't have a season number, but this show appears to have only 1 season, setting season number to 1...
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Show information parsed from file name
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (release_group): HorribleSubs
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (air_date): None
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (ab_episode_numbers): [1]
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (episode_numbers): []
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (season_number): None
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result (series_name): Cardcaptor Sakura Clear Card
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Parse result not sufficient (all following have to be set). Won't save release name
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Using cached parse result for [HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Analyzing name: [HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv
2018-01-07 11:26:19 INFO     Thread-32 :: [7b9111e] Processing A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p]\[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Post-processing video files: [u'[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv']
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Post-processing files: [u'[HorribleSubs] Cardcaptor Sakura Clear Card - 01 [1080p].mkv']
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Processing folder: A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p]
2018-01-07 11:26:19 DEBUG    Thread-32 :: [7b9111e] Processing path: A:\NZBs\Holding Tank\Anime\[HorribleSubs].Cardcaptor.Sakura.Clear.Card.-.01.[1080p]
```
I also can't overwrite that.  When I try to edit the 'Scene Absolute' column, I'm faced with a message that says 'Unable To Retrieve Episode'.  The debug log only states this when that event happens:
```
2018-01-07 11:23:59 DEBUG    Thread-33 :: [7b9111e] 73954: No entries found in show: Card Captor Sakura: Clear Card
2018-01-07 11:23:49 DEBUG    Thread-32 :: [7b9111e] 73954: No entries found in show: Card Captor Sakura: Clear Card
2018-01-07 11:23:25 DEBUG    Thread-33 :: [7b9111e] 73954: No entries found in show: Card Captor Sakura: Clear Card
```
Now for AniDB... It doesn't work and I have no logs to show.  Tinkering last night I apparently got API banned from AniDB for bad login info, I can still login via web though, but since I've seen nothing in the logs for AniDB.  If I enable or disable it, nothing appears in the debug logs.  I don't see AniDB listed as an indexer and I'm baffled as to why nothing about it is even mentioned in the logs.  I realize that without logs it's not very useful to ask for help but AniDB has kinda STOPPED turning up in my logs so I have no idea what logs to provide.

**Comment by abeloin** [comment]
@AshleyAshes It should now work(in master) with theTVDB, can you try it?

As for TVMaze and TMDB, I'm currently looking as to why it wouldn't PP it.

**Comment by AshleyAshes** [comment]
I'm sorry, can you clarify WHAT should work?  I've been doing a lot of manual work with Medusa to keep CCS downloading and so I'm a bit confused since this is a mutlifaceted issue.

I do see that The TVDB has a separate 'Cardcaptor Sakura Clear Card' listing on TVDB and Medusa see's it, I think that this is against TVDB's rules.  It has no thumbnails, descriptions and not the most recent episodes are listed.  This seems like something a user added and TVDB admins will nuke once they find out.

I've ADDED that listing into my Medusa to test it and it fails to find search results.

As for the Cardcaptor Sakura listing, this is very broken.   Firstly, it's mistaking Season 2 episodes, as in those from the 1990's as being Clear Card episodes.  So I already have the original series and it's misreading those.  Then I download and manually rename episodes to absolute numbers and it'll delete OLD episodes that it thinks are NEW episodes due to the numbering confusion.  (And this is WITH absolute numbering enabled or marking older episodes as 'archived')

I realize this is not the dev's fault, it's the unique circumstances of this series, but in it's current state, CCS is pretty broken.

**Comment by p0psicles** [comment]
We have season scene exceptions for these. The exceptions (mappings) are added on thexem.de.

For anime we currently only support tvdb. As we heavily depend in the exceptions which are only available for tvdb.

If you have an anime show on tvmaze or tmdb that follows the scene exactly, you could try it. But those are rare.

**Comment by abeloin** [comment]
@AshleyAshes Sorry for the late reply.
There was an issue where with thexem.de not being applied properly in certain show in Medusa and "Clear Card" was one of them.

A fix was pushed in commit [d9b688fb423bac5e0daceeedca66ae44b6f447d3](https://github.com/pymedusa/Medusa/commit/d9b688fb423bac5e0daceeedca66ae44b6f447d3)
which was included in [release 0.2.1](https://github.com/pymedusa/Medusa/releases/tag/v0.2.1)
The serie tested was this one https://www.thetvdb.com/?tab=series&id=70668

If the new episode(season 4) still show in season 2, the xem data for the series might be incorrect in Medusa and you try a forec full update.

**Comment by p0psicles** [comment]
Closing as issue seems to be resolved. As there has not been a response for a while.
